### DISCUSSION TITLE
error updating collection

### DISCUSSION THREAD

**Discussion by brunowilliang** [pocketbase/pocketbase#1316] [Q&A]
I'm trying to update my profile collection and I'm getting this error... does anyone know how to solve it?

![Screen Shot 2022-12-20 at 11 16 34](https://user-images.githubusercontent.com/41432581/208688069-9d1f4c4b-188a-48df-aa98-0569d5978ff8.png)
![Screen Shot 2022-12-20 at 11 16 22](https://user-images.githubusercontent.com/41432581/208688071-37ec16dc-0df5-4b95-ab1b-f91cb5e05c06.png)

**Comment by ganigeorgiev** [comment]
Are you sure that `data?.id` is set?

To debug it you have 3 options:
- if the request is executed client-side, check the Request and Response data in the browser's "Inspect tool > Network".

- start PocketBase with `--debug`

- check the 404 detailed log message from the Admin UI > Logs

**Reply by brunowilliang** [reply]
@ganigeorgiev but I'm using fly.io, how would I use it in debug mode?

**Reply by ganigeorgiev** [reply]
You can pass the flag in the docker `CMD` instruction, aka:
```
CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8080", "--debug"]
```
The logs will be logged in the console from the fly.io dashboard.

But that is not necessary. You can try the other 2 options. I suspect that `data?.id` is not set and you are sending undefined. Another thing that also could be a reason for the 404 is that your collection's "Update API rule" is not satisfied (you can try temporary changing it to an empty string to allow everyone access).

**Comment by brunowilliang** [comment]
@ganigeorgiev my data?.id it returns me first undefined and then the id...

![Screen Shot 2022-12-20 at 11 39 37](https://user-images.githubusercontent.com/41432581/208693912-6c17d3db-624f-4a55-ac00-469a381a9b38.png)

**Reply by ganigeorgiev** [reply]
I'm not sure how the `data` object is constructed.

It seems that something is triggering your `updateProfile()` method with empty data.

**Reply by brunowilliang** [reply]
@ganigeorgiev 
I can register the data, the problem is that if I open the screen and there is no data, it gives this wa
![Simulator Screen Shot - iPhone SE (3rd generation) - 2022-12-20 at 16 11 14](https://user-images.githubusercontent.com/41432581/208747607-4e2f32e3-6fee-43cd-99ee-020e80d7a58e.png)
rning;
![Simulator Screen Shot - iPhone SE (3rd generation) - 2022-12-20 at 16 09 45](https://user-images.githubusercontent.com/41432581/208747597-2a723392-fcef-4bf7-a205-668bbfea8439.png)

I'm passing autoCancel, but it doesn't seem to work
![code2](https://user-images.githubusercontent.com/41432581/208747751-d4154ec6-b3b7-4138-99d2-7c0373655d4b.png)

**Reply by ganigeorgiev** [reply]
@Brunowilliang Autocancellation has nothing to do with this and I don't think this is related to the SDK. I'm not familiar with your codebase. Something is triggering your `updateProfile()` method with empty `data` argument but I don't know when, where and how this `data` is generated and why this method is called.
